### PR TITLE
MEAT SPIKE EXPANSION

### PR DESCRIPTION
**Pull request by harmonyn**
- Meat spikes are now constructable! A frame requires 5 metal (same as computer and machine frames), add 4 rods for spikes. Spikes can be removed, same as bookshelves.
- Meat spike frames can be wrenched and moved. Have fun lining your rage cage in ACTUAL SPIKES.
- You can slam humans that you are grabbing into meat spikes, causing 20 damage per slam. THIS NUMBER CAN ABSOLUTELY BE ADJUSTED IF NECESSARY, originally I had it at 15 but coderbus feedback said make it more
- You can use corgis on meat spikes, producing corgi hide and meat. You absolute monster. (This is a codersprite let me know if it sucks too much)

corgi
- Removed tremendous amounts of copypasta from meatspike code.

### CODE REVIEW

**Comment by Xhuis** [comment]
\>you can put corgis on meat spikes
https://www.youtube.com/watch?v=qQBO4tbvk_U

**Comment by AlexCrimson** [comment]
Time to build one in the HoPs office whilst he is out being HoPSec so he has a nice surprise when he gets back. Poor Ian.

Should add cats and spiders to the to-do list.

**Comment by Supermichael777** [comment]
can you make the hide the last thing to come off the hide to mach the visuals. also what does deconstructing it while it is in use do, is that handed?

**Comment by bgobandit** [comment]
A) sure but I'd have to do it for monkey, xeno and bear as well

B) HONK LET ME FIX THAT

**Comment by Aranclanos** [comment]
don't copypaste code

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -366,7 +366,7 @@ Class Procs:
 		return 1
 	return 0
 
-/obj/machinery/proc/default_unfasten_wrench(mob/user, obj/item/weapon/wrench/W, time = 20)
+/obj/proc/default_unfasten_wrench(mob/user, obj/item/weapon/wrench/W, time = 20)
```
Comment (Anonus) [inline code/game/machinery/machinery.dm]: this should probably be in object procs somewhere, instead of in machinery.dm if it's not machinery only now

Comment (bgobandit) [inline code/game/machinery/machinery.dm]: SEE: THE ABSOLUTE HEIGHT OF INCONSISTENT STUPIDITY

Comment (Iamgoofball) [inline code/game/machinery/machinery.dm]: here's why you're wrong about this being stupid:

default_unfasten_wrench() is only called on objects that have a call to them(such as machinery)

so not everything has it

File: code/game/objects/structures/kitchen_spike.dm [hunk 2]
```diff
@@ -2,13 +2,36 @@
...
+/obj/structure/kitchenspike_frame/attackby(obj/item/I, mob/user, params)
+	add_fingerprint(user)
+	if(default_unfasten_wrench(user, I))
+		return
+	else if(istype(I, /obj/item/stack/rods))
+		var/obj/item/stack/rods/R = I
+		if(R.get_amount() >= 4)
+			R.use(4)
+			user << "<span class='notice'>You add spikes to the frame.</span>"
+			var/obj/F = new /obj/structure/kitchenspike(src.loc,)
```
Comment (Iamgoofball) [inline code/game/objects/structures/kitchen_spike.dm]: you know

this isnt for this PR but

standardizing the "make shit, add fingerprints, qdel old shit" into a proc might be nice

you dont have to do it here tho

File: code/game/objects/structures/kitchen_spike.dm [hunk 3]
```diff
@@ -25,55 +48,58 @@
...
+					src.icon_state = "spikebloody"
+				if(istype(G.affecting, /mob/living/carbon/alien))
+					src.icon_state = "spikebloodygreen"
+					src.meattype = MEATTYPE_ALIEN
+					src.skintype = SKINTYPE_ALIEN
+				if(istype(G.affecting, /mob/living/simple_animal/hostile/bear))
+					src.icon_state = "spikebloodybearz"
+					src.meattype = MEATTYPE_BEAR
+					src.skintype = SKINTYPE_BEAR
+				if(istype(G.affecting, /mob/living/simple_animal/pet/dog/corgi))
```
Comment (duncathan) [inline code/game/objects/structures/kitchen_spike.dm]: oh my god four if(istype())s in a row I am having horrifying flashbacks to pipe construction code

File: code/game/objects/structures/kitchen_spike.dm [hunk 4]
```diff
@@ -84,57 +110,51 @@
...
-				new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/bear(src.loc )
-				usr << "<span class='notice'>You remove some meat from the bear.</span>"
-			else if(src.meat == 1)
-				src.meat--
-				new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/bear(src.loc)
-				usr << "<span class='notice'>You remove the last piece of meat from the bear.</span>"
-				src.icon_state = "spike"
-				src.occupied = 0
+		if (src.meat > 1)
+			switch(src.meattype)
```
Comment (duncathan) [inline code/game/objects/structures/kitchen_spike.dm]: oh god a switchcase for each possible type of meat more flashbacks aaaaaa

File: code/game/objects/structures/kitchen_spike.dm [hunk 5]
```diff
@@ -84,57 +110,51 @@
...
+				if(MEATTYPE_ALIEN)
+					new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/xeno(src.loc )
+				if(MEATTYPE_BEAR)
+					new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/bear(src.loc )
+				if(MEATTYPE_CORGI)
+					new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/corgi(src.loc )
+			src.meat--
+			usr << "<span class='notice'>You remove some meat from [src].</span>"
+		else if (src.meat == 1)
+			switch(src.meattype)
```
Comment (duncathan) [inline code/game/objects/structures/kitchen_spike.dm]: see above

File: code/game/objects/structures/kitchen_spike.dm [hunk 6]
```diff
@@ -84,57 +110,51 @@
...
+					new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/monkey(src.loc)
+				if(MEATTYPE_ALIEN)
+					new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/xeno(src.loc)
+				if(MEATTYPE_BEAR)
+					new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/bear(src.loc)
+				if(MEATTYPE_CORGI)
+					new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/corgi(src.loc)
+			src.meat--
+		else if(src.skin >=1)
+			switch(src.skintype)
```
Comment (duncathan) [inline code/game/objects/structures/kitchen_spike.dm]: see above

File: code/game/objects/structures/kitchen_spike.dm (hunk 4 above)
Comment (duncathan) [inline code/game/objects/structures/kitchen_spike.dm]: you can probably address the [x]type comments I made by making the MEATTYPE and SKINTYPE defines paths; see __DEFINES/pipe_construction.dm to see what I mean by that, and game/machinery/pipe/construction.dm for how it was implemented

File: code/game/objects/structures/kitchen_spike.dm (hunk 3 above)
Comment (bgobandit) [inline code/game/objects/structures/kitchen_spike.dm]: if you think this is bad look at the code it used to be

File: code/game/objects/structures/kitchen_spike.dm (hunk 4 above)
Comment (bgobandit) [inline code/game/objects/structures/kitchen_spike.dm]: they 

ARE

define paths

do you even read the code

**Comment by PKPenguin321** [comment]
>Have fun lining your rage cage in ACTUAL SPIKES.

that's metal as fuck and i love it

**Comment by Lumipharon** [comment]
@bgobandit Gotta make all the animals spikeable (cat,spiders,mining mobs/etc)

**Comment by Jalleo** [comment]
@Lumipharon even butterflies and drones?

**Comment by bgobandit** [comment]
sprite hides for them and I will do so

**Comment by Razharas** [comment]
MUH MEAT SPIKE

**Comment by Kromgar** [comment]
We need a meat spike for goliaths.

**Comment by Bluespace** [comment]
I think it's time humans are allowed to be meat spiked once dead. 
Rage cage with fallen foes lining the edge?

**Comment by Miauw62** [comment]
meat spikes instantly destroy the body, which is what the gibber is for.

File: code/game/machinery/machinery.dm (hunk 1 above)
Comment (optimumtact) [inline code/game/machinery/machinery.dm]: this is retarded

File: code/game/objects/structures/kitchen_spike.dm [hunk 7]
```diff
@@ -22,58 +44,83 @@
...
+		"alien"		= MEATTYPE_ALIEN,
+		"bear"		= MEATTYPE_BEAR,
+		"corgi"		= MEATTYPE_CORGI,
+	)
+	var/global/list/skinlist = list(
+		"monkey"	= SKINTYPE_MONKEY,
+		"alien"		= SKINTYPE_ALIEN,
+		"bear"		= SKINTYPE_BEAR,
+		"corgi"		= SKINTYPE_CORGI,
+	)
```
Comment (phil235) [inline code/game/objects/structures/kitchen_spike.dm]: three global lists only used once in meat spike code, no. Especially not with the generic name "allowed_types". Make these list vars of kitchenspike instead.

Comment (duncathan) [inline code/game/objects/structures/kitchen_spike.dm]: What? They are vars of kitchenspike. Only kitchenspikes have these lists.

Comment (Miauw62) [inline code/game/objects/structures/kitchen_spike.dm]: >var/global/list
>not global

makes sence

Comment (duncathan) [inline code/game/objects/structures/kitchen_spike.dm]: The global keyword declares a variable global to all objects of that type. There's only one of each of these lists in the world, but only kitchenspikes have the list as a var.

Comment (duncathan) [inline code/game/objects/structures/kitchen_spike.dm]: If you want a variable that's completely global, you just declare it with no indentation. Lists like this are what global is for.

Comment (optimumtact) [inline code/game/objects/structures/kitchen_spike.dm]: http://www.byond.com/docs/guide/chap05.html section 4 covers this

Comment (Cheridan) [inline code/game/objects/structures/kitchen_spike.dm]: dunc is right: http://www.byond.com/docs/guide/chap05.html
subheading 4

Comment (optimumtact) [inline code/game/objects/structures/kitchen_spike.dm]: thanks cheridan

**Pull request merged**